src/app/machines/views/MachineDetails/MachineHeader/MachineHeader.tsx

```diff
@@ -52,6 +55,10 @@ const MachineHeader = ({
   const statuses = useSelector((state: RootState) =>
     machineSelectors.getStatuses(state, systemId)
   );
+  const { dispatch: dispatchForSelectedMachines } =
+    useSelectedMachinesActionsDispatch({
+      selectedMachines: { items: [systemId] },
+    });
```

Comment:
```suggestion
    });

  const handleActionClick = (action: NodeActions) => {
    sendAnalytics(
      "Machine details action form",
      getNodeActionTitle(action),
      "Open"
    );
    const isImmediateAction =
      action === NodeActions.LOCK || action === NodeActions.UNLOCK;

    if (isImmediateAction) {
      dispatchForSelectedMachines(machineActions[action]);
    } else {
      const view = Object.values(MachineHeaderViews).find(
        ([, actionName]) => actionName === action
      );

      if (view) {
        setSidePanelContent({ view });
      }
    }
  };
```

Reply:
Provided you commit the suggestion above you should be able to use a single handler for both mobile and desktop action menus.